lib/rubocop/cop/lint/redundant_initialize.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for `initialize` methods that are redundant.
+      #
+      # An initializer is redundant if it does not do anything, or if it only
+      # calls `super` with the same arguments given to it. If the initializer takes
+      # an argument that accepts multiple values (`restarg`, `kwrestarg`, etc.) it
+      # will not register an offense, because it allows the initializer to take a different
+      # number of arguments as its superclass potentially does.
+      #
+      # NOTE: If an initializer argument has a default value, RuboCop assumes it
+      # to *not* be redundant.
+      #
+      # @example
+      #   # bad
+      #   def initialize
```

Comment:
I'm wondering if there is a situation where someone would intentionally define an empty `initialize` to prevent the parent's `initialize` being called (even though this is probably a terrible approach!).

Reply:
Maybe it should only flag an empty initialize if there's no superclass?

Reply:
@bbatsov any thoughts? Are you concerned about this edge case?

Reply:
This seems to spill over into the domain of the [Lint/MissingSuper](https://docs.rubocop.org/rubocop/cops_lint.html#lintmissingsuper) cop. I can imagine (rare) but legit use cases for omitting `super`, e.g. for abstract base classes:

```ruby
class Parent
  def initialize
    raise NotImplementedError, "I'm abstract!"
  end
end

class Child < Parent
  def initialize
     # we deliberately don't call super here
  end
end
```

Reply:
@dvandersluis I'm not very concerned about this. I'd mention this case in the cop description, though. We can adjust this based on the feedback we get from our users once this is out.

Reply:
Updated as per @bbatsov.